apps/verticalhome/js/pin_app_navigation.js

```diff
@@ -0,0 +1,294 @@
+'use strict';
+/* global ApplicationSource */
+/* global BookmarkSource */
+/* global CollectionSource */
+/* global configurator */
+/* global dispatchEvent */
+/* global GaiaGrid */
+
+(function(exports) {
+
+
+  /**
+   * This is PinAppNavigation's constructor.
+   * @param {Object} list [DOM container that contains pin elements]
+   */
+  function PinAppNavigation(list) {
+    if (!list && typeof list !== 'object') {
+      console.log('List of pin apps must be an object');
+      return;
+    }
+
+    this.elemList = list;
+    window.removeEventListener('keydown', this);
+    window.addEventListener('keydown', this);
+  };
+
+
+  PinAppNavigation.prototype = {
+
+    selector: '',
+    elements: null,
+    preventDef: false,
+    selectedElemIndex: 0,
+    elemList: null,
+    prevVerticalKey: null,
+
+    middleElem: false,
```

Comment:
It is almost never right to define properties like these in the prototype. Put methods in the prototype, but define these properties in the constructor, please.  Or, if you've got a very good reason why these should be in the prototype, add a comment to explain that reason.